src/ConnectionDispatcher.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <memory>
+
+#include "DBConnection2.hpp"
+
+class ConnectionDispatcher {
```

Comment:
@Morheit ,
1) I don't see the reason for storing ptr in smart ptr. you can store just ptr.
2) see http://qaru.site/questions/89999/what-is-single-and-double-dispatch. I don't see the point to store ptr at all. In the sample you can see just sequence of calls and that's it. There is just "magic" with parameters and thats it. Try to implement something like this. Consider passing lambdas to dispatcher to pass custom hander.

Reply:
@malirod smart ptr replaced by ptr